### DISCUSSION TITLE
Allow customization of the axios instance

### DISCUSSION THREAD

**Discussion by javoscript** [laravel/nova-issues#4689] [Ideas & Feature Requests]
I was trying to add a custom header to all requests being made from Nova, and found out that the `Nova.request()` function, that provides a Nova-customized instance of axios, doesn't allow for any further customization.

I ended up wrapping the function with an implementation of my own (leaving it here for anyone facing the same "issue"):

```js
Nova.booting(_app => {
    let oldNovaRequest = Nova.request

    Nova.request = (options) => {
        let axiosInstance = oldNovaRequest(options)

        // modify the Nova axios instance
        axiosInstance.interceptors.request.use(req => {
            req.headers['some-custom-header'] = 'custom-header-value'
            return req
        })

        return axiosInstance
    }
})
```

I think it would be really useful if the provided `app` object in the `booting()` callback provided access to an axios instance being used by Nova and allowed for further customization 😊.

**Comment by davidhemphill** [comment]
What kind of customizations are you adding and why?

**Reply by javoscript** [reply]
I ended up solving my specific situation in another way. I'm implementing several profiles per user, and wanted to have only one at a time as the "active" one, and being able to change profiles, all in the Nova admin panel.

In the end, I implemented it all in the backend, with a custom middleware and setting some data in the session.

But through the process of investigating an implementation, I got to a point where I wanted to customize the `axios` instance and wasn´t able in any other way than the shown in the original post.

Nothing comes to mind right now, but being able to change the axios config, or implement interceptors will surely come in handy for some use cases.
